src/YDialogSpy.cc

```diff
@@ -514,7 +514,12 @@ void YDialogSpy::exec()
 
             continue;
         }
-        else if ( event->widget() == priv->upButton ) priv->moveSelectedUp();
+
+        // check for NULL which is generated for timeout events which can be
+        // triggered by the original dialog
+        if (!event->widget()) continue;
```

Comment:
It does help, but for the wrong reason: The segfault was caused by a YTimeoutEvent which inherits the default virtual YEvent::widget() method that always returns 0.

There should be another check right after this one that checks if that widget's dialog is the YDialogSpy:

    if ( event->widget()->findDialog() == this ) continue;